ReactNativeClient/lib/joplin-renderer/MdToHtml/rules/video.js

```diff
@@ -0,0 +1,61 @@
+// LOGIC: Extract the & UUID of the youtube video and render it in iframe
+const videoCss = () => {
+	return [
+		{
+			inline: true,
+			mime: 'text/css',
+			text: `
+                .video-container {
+                    position: relative;
+                    width: 100%;
+                    height: 0;
+                    padding-bottom: 56.25%;
+                }
+                .video-iframe {
+                    position: absolute;
+                    top: 0;
+                    left: 0;
+                    width: 100%;
+                    height: 100%;
+				}`,
+		},
+	];
+};
+
+function addContextAssets(context) {
+	if ('video' in context.pluginAssets) return;
+
+	context.pluginAssets['video'] = videoCss();
+}
+
+function installRule(markdownIt, mdOptions, ruleOptions, context) {
+	const defaultRender = markdownIt.renderer.rules.link_open;
+	const videoIDRegex = /^.*(youtu\.be\/|v\/|u\/\w\/|embed\/|watch\?v=|&v=)([^#&?]*).*/;
```

Comment:
This regex is faulty, for examples it matches this, but shouldn't:

`'https://www.a.com/watch?v=SADub7W22Zg'.match(/^.*(youtu\.be\/|v\/|u\/\w\/|embed\/|watch\?v=|&v=)([^#&?]*).*/)`

Reply:
Sure, we can remove this bug by checking if the domain name is youtube/youtu.be
``` js
	const key = /.*(youtube|youtu.be).*/im;
	const match = link.match(key);
if (videoID[2] && match[1]==='youtube') {...}
```

Reply:
How about looking at a url parsing lib? Or maybe there's already a function for this in Joplin?

Reply:
Okay, I'll look into it and let you know!

Reply:
You can use the `url` package, which is already available to all clients: https://www.npmjs.com/package/url

Reply:
yea I looked into it, and we currently use this dependency in Joplin too, this update got delayed a bit because I was busy trying to implement [this ](https://discourse.joplinapp.org/t/add-tables-to-the-toolbar/6213/15?u=rishabh.malhotra) feature request. I will get this done ASAP